### DISCUSSION TITLE
Is singleton supervision planned?

### DISCUSSION THREAD

**Discussion by databrecht** [slawlor/ractor#194] [General]
Is something like: https://elixirforum.com/t/singleton-global-supervised-singleton-processes-for-elixir/2175 planned or is there another way planned to make sure in a cluster that actors don't run twice on multiple nodes or get shared? (I'm not terribly familiar with the BEAM engine so I'm not sure whether what I asked makes perfect sense)

**Comment by slawlor** [comment]
I don't have plans for anything like this in particular, but it should be doable likely by cloning similar logic from that elixir forum. You would need some type of supervisor that runs on each node that's monitoring the state of these "global" processes and they need to coordinate to determine if one process is running somewhere and probably some type of distributed lock mechanism such that you don't accidentally start multiple.

I think it's very feasible, but I don't have a use case in mind. It could definitely be something you could push to ractor_actors which is my utility actors crate that I'm building. It's not something I would envision being in the base library however, even in the post you linked it's an additional dependency
